src/main/java/com/exasol/projectkeeper/validators/changelog/ChangelogFileGenerator.java

```diff
@@ -0,0 +1,33 @@
+package com.exasol.projectkeeper.validators.changelog;
+
+import static com.vdurmont.semver4j.Semver.SemverType.LOOSE;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.vdurmont.semver4j.Semver;
+
+/**
+ * This class generates the content for the changelog file.
+ */
+class ChangelogFileGenerator {
+    private static final String NL = System.lineSeparator();
+
+    /**
+     * Generate the content for the changelog file.
+     * 
+     * @param versions list of project versions
+     * @return content for the changelog file
+     */
+    String generate(final List<String> versions) {
+        final List<String> sortedVersions = versions.stream().map(version -> new Semver(version, LOOSE))
+                .sorted(Comparator.reverseOrder()).map(Semver::getValue).collect(Collectors.toList());
+        final StringBuilder templateBuilder = new StringBuilder();
+        templateBuilder.append("# Changes" + NL + NL);
```

Comment:
In some project, we start from some later version. For example, we start using changelog from 3.0.0. Would it be a problem if versions below 3.0.0 are missing in the changelog file?

Reply:
This validator requires that all changes_x.x.x.md files that exist are listed in the changelog.md file. So if we did not have changes files for those earlier versions it's not a problem. If we had, we can also list them.